### ISSUE TITLE
Quote Block text alignment conflicts with added position alignment support

### ISSUE THREAD

**Issue by grappler** [WordPress/gutenberg#34454]
### Description

I have added [align support](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#align) to the quote block. The code that I used can be found below.

This conflicts with the [`align` attribute](https://github.com/WordPress/gutenberg/blob/7901895ca20cf61e402925e31571d659dab64721/packages/block-library/src/quote/block.json#L25-L27) in the quote block which is used for text alignment with the block.

I would have expected for the attribute to be named `textAlign` instead of `align`.

### Step-by-step reproduction instructions

1.  Add align support to quote block
```php
import { addFilter } from '@wordpress/hooks';
const enableAlignmentOnBlocks = [  'core/quote' ];
// Add alignments to group block.
const addAlignmentToBlocks = ( settings, name ) => {
	if ( ! enableAlignmentOnBlocks.includes( name ) ) {
		return settings;
	}

	return {
		...settings,
		supports: {
			...settings.supports,
			align: [ 'left', 'center', 'right' ],
		},
	};
};
addFilter( 'blocks.registerBlockType', 'mcs-blocks/alignments/group-block', addAlignmentToBlocks );
```
2. Add the [quote pattern](https://wordpress.org/patterns/pattern/quote/)
3. Get an error "Block validation: Block validation failed for `core/quote`" and changing the alignment changes both the position and text alignment

### Environment info

- WordPress 5.8/5.7.2

### Pre-checks

- [X] I have searched the existing issues.
- [X] I have tested with all plugins deactivated except Gutenberg.

**Comment by youknowriad** [comment]
Indeed, I'm not sure that the Quote block is the only one in this situation. This is unfortunately "historic". Meaning when these attributes were added, block supports didn't exist yet and they evolved organically after that. We could consider renaming the existing attribute which would require a deprecated version I think. We may have to check what other blocks do the same.

**Comment by ellatrix** [comment]
Cc @t-hamano

**Comment by t-hamano** [comment]
The current `align` attribute, which actually means text alignment, needs to be migrated to [the new `textAlign` block support](https://make.wordpress.org/core/2024/06/24/miscellaneous-editor-changes-in-wordpress-6-6/#Added-Text-alignment-block-support).

Tracking Issue: https://github.com/WordPress/gutenberg/issues/60763

The new `textAlign` block support should no longer conflict with the `align` block support, since it's saved as a style instead of an attribute.

But we need to first resolve the style override issue that this PR attempts to address: https://github.com/WordPress/gutenberg/pull/62260

**Comment by tyxla** [comment]
This appears to be resolved now, with all items @t-hamano mentioned resolved too. 

#42960 renamed the text alignment attribute to `textAlign` (with a deprecation that migrates old `align` values), and #64188 added native `align` support for block positioning, so the two no longer collide and the filter isn't needed anymore. 

Closing as completed.
